### PR TITLE
Close template window if it wasn't open before the run

### PR DESCRIPTION
**Pull request by huysentruitw**
This solves the annoyance of having all .tt files open after each build. Template files that were previously closed will get closed again in the editor.

### CODE REVIEW

**Comment by bennor** [comment]
What version of VS are you using? I haven't had this reported before.

File: AutoT4/ExtensionMethods.cs [hunk 1]
```diff
@@ -49,9 +49,8 @@ public static void RunTemplate(this ProjectItem template)
             }
             else
             {
-                if (!template.IsOpen)
-                    template.Open();
-                template.Save();
+                if (template.IsOpen) template.Save();
```
Comment (bennor) [inline AutoT4/ExtensionMethods.cs]: Can you put the statements indented on the next line please.

**Comment by huysentruitw** [comment]
I've tested this with VS2015, will update indentation right away.

**Comment by huysentruitw** [comment]
Oh, btw, a part of our .tt files are solution items and not project items, maybe that was the error/edge case you couldn't remember :)

File: AutoT4/ExtensionMethods.cs [hunk 2]
```diff
@@ -49,10 +49,16 @@ public static void RunTemplate(this ProjectItem template)
             }
             else
             {
-                if (template.IsOpen)
+                if (!template.IsOpen)
+                {
+                    var window = template.Open();
```
Comment (bennor) [inline AutoT4/ExtensionMethods.cs]: Do we need to dispose this?

**Comment by bennor** [comment]
I think you're right on the fact that they're solution items - that would explain there being no project item for them.

File: AutoT4/ExtensionMethods.cs (hunk 2 above)
Comment (huysentruitw) [inline AutoT4/ExtensionMethods.cs]: `Window` does not implement `IDisposable`

**Comment by bennor** [comment]
Thanks mate. I have one other bug to look into, and that could take a while, so I'll aim to have this released by early next week

**Comment by huysentruitw** [comment]
Hi Bennor, checked it today, but the AutoT4 package in the VisualStudio gallery is still the version before this fix. Is it possible to get it released or are you still waiting for that other bug-fix?

**Comment by bennor** [comment]
Sorry. I hit a roadblock with the other change and while I was waiting on a response to my question in the VS extensions Gitter, I forgot about this. Will get it released over the Easter break -- the other fix can wait (it may not be possible anyway).

**Comment by huysentruitw** [comment]
No problem, just wanted to send you some reminder ;) Thanks for considering the release. The strange thing is, if I install the own-built extension, then the tabs are closed again, but after some time (maybe VS restart) the issue comes back as if VS has overridden the extension with the version from the gallery and I need to reinstall the own-built version. Is this something you're aware of?

**Comment by bennor** [comment]
Not something I'm aware of, but it sort of makes sense. 

I've updated the version in the gallery now anyway, so you should be good to use that now. Thanks again for the PR! 😄

**Pull request merged**